code/modules/antagonists/changeling/powers/absorb.dm

```diff
@@ -67,9 +67,16 @@
 
 		//Some of target's recent speech, so the changeling can attempt to imitate them better.
 		//Recent as opposed to all because rounds tend to have a LOT of text.
-		var/list/recent_speech = list()
 
-		var/list/say_log = target.logging[LOG_SAY]
+		var/list/recent_speech = list()
+		var/list/say_log = list()
+		var/log_source = target.logging
+		for(var/log_type in log_source)
+			var/nlog_type = text2num(log_type)
+			if(nlog_type & LOG_SAY)
+				var/list/reversed = log_source[log_type]
+				if(islist(reversed))
+					say_log = reverseRange(reversed.Copy())
 

```

Comment:
Shouldn't your `break` after this?

Reply:
Or make it additive?

Reply:
Good point, `break` would do it since after it locates the LOG_SAY it doesn't need to keep looking